### PR TITLE
new: keep window after deleting buffer; 

### PR DESCRIPTION
**Pull request by skywind3000**
- Keep window layout while deleting buffer (`let g:bufferhint_KeepWindow = 1`, default is 0)
- Prompt `No write since last change` and exit while attempting delete a modified buffer. Otherwise it will fail into an error state: `bdelete (without !)` failed but `remove(bids, a:idx)` succeeded (in KillByIndex).
- Syntax highlight settings may lost by using `hi clear` in Popup(), which make my LineNr (color of line numbers) setting invalid.

### CODE REVIEW

File: plugin/bufferhint.vim [hunk 1]
```diff
@@ -97,7 +101,7 @@ fu! bufferhint#Popup()
         sy clear
         sy match KeyHint /^../
         sy match AtHint /@/
-        hi clear
+        "hi clear
```
Comment (bsdelf) [inline plugin/bufferhint.vim:104]: `hi clear` => `hi clear KeyHint`

Could you please test if this piece of code works in your application scenario?

Comment (skywind3000) [inline plugin/bufferhint.vim:104]: LineNr highlight seems correct after changing to 'hi clear KeyHint', need I make another pull request with it ?
